src/Uno.UI.Composition/Composition/BorderVisual.skia.cs

```diff
@@ -264,30 +265,40 @@ private unsafe void UpdateBackgroundPath(bool useInnerBorderBoundsAsAreaForBackg
 			roundRect.Handle,
 			&rect,
 			useInnerBorderBoundsAsAreaForBackground ? innerRadii : outerRadii);
-		_backgroundPath.AddRoundRect(roundRect);
-		_backgroundPath.Close();
+		backgroundPath.AddRoundRect(roundRect);
+		backgroundPath.Close();
+
+		// Unfortunately, this will cause an unnecessary render invalidation
+		((CompositionPathGeometry)_backgroundShape!.Geometry!).Path = new CompositionPath(new SkiaGeometrySource2D(backgroundPath));
```

Comment:
There shouldn't be any "users" here. The `BackgroundPath` is private to the BorderVisual (but was marked internal in another PR because it was needed for BackgroundTransition) and it not exposed to the user. If somehow the `Geometry` is not CompositionPathGeometry, we should indeed crash.

Reply:
I made it private now.